### ISSUE TITLE
Search with umlauts results in 0 search results

### ISSUE THREAD

**Issue by Sereghon** [sphinx-doc/sphinx#10649]
### Describe the bug

When the search contains umlauts, I get no search results, despite the fact, that pages with ä,ö and ü exists.

### How to Reproduce

Simply add a word which contains umlauts and try to find it in your docu.

### Expected behavior

Search is working with umlauts as well.

### Your project

-

### Screenshots

![grafik](https://user-images.githubusercontent.com/19366697/177948994-7f89ce89-0b31-45ad-916d-ec43b654ccc9.png)

### OS

Windows 10

### Python version

3.9.7

### Sphinx version

4.5.0

### Sphinx extensions

sphinx.ext.autodoc, sphinx.ext.doctest, sphinx.ext.intersphinx, sphinx.ext.todo, sphinx.ext.coverage, sphinx.ext.imgmath, sphinx.ext.ifconfig, sphinx.ext.viewcode, sphinx.ext.graphviz, sphinx.ext.extlinks

### Extra tools

Mozilla Firefox

**Comment by TimKam** [comment]
Can you provide an example project? I can't reproduce the issue with the latest Sphinx version using what I would expect to be a minimal example and assume it may be an encoding issue that you can fix in your setup.

**Comment by Sereghon** [comment]
Hi @TimKam,

I've attached one for you.

[sphinx-example.zip](https://github.com/sphinx-doc/sphinx/files/9084690/sphinx-example.zip)

**Comment by TimKam** [comment]
I still did not manage to reproduce the error, even when using your project and Sphinx 4.5.0. Would you mind sharing your build result? I suspect this could be an encoding issue.

**Comment by konstantinbo** [comment]
Yes, of course: [sphinx-example_builds.zip](https://github.com/sphinx-doc/sphinx/files/9130841/sphinx-example_builds.zip)

Also this is my current build environment: 

- OS: Windows 10 Pro, 21H2 Build 19044.1826
- Python 3.9.7

- `PATH=C:\Python39\;C:\Python39\Scripts\;C:\Program Files\AdoptOpenJDK\jdk-11.0.10.9-hotspot\bin;C:\Program Files\Zulu\zulu-8\bin\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\PuTTY\;C:\cygwin64\bin;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Program Files\TortoiseGit\bin;C:\Program Files\LLVM\bin;C:\Program Files\TortoiseSVN\bin;C:\Users\Konstantin\.cargo\bin;C:\Python36\Scripts\;C:\Python36\;C:\Users\Konstantin\AppData\Local\Microsoft\WindowsApps;C:\Users\Konstantin\AppData\Local\Programs\Git\cmd;C:\Users\Konstantin\AppData\Local\Programs\Microsoft VS Code\bin;C:\texlive\2021\bin\win32;C:\Users\Konstantin\AppData\Local\JetBrains\Toolbox\scripts`

**Comment by Sereghon** [comment]
Hello @TimKam , did you found time to take a look into it?

**Comment by Sereghon** [comment]
Hello @TimKam , did you found time to take a look into it?

**Comment by Sereghon** [comment]
Hello
I've just noticed that the search for umlauts is working now. 
We updated sphinx version from 4.5.0 to 5.3.0 on 2022-10-21.

Atleast for me this issue can be closed. If something occurs in the future I will raise a new request with updated information.

Best regards
